include/verilated.cpp

```diff
@@ -3098,3 +3098,17 @@ void VerilatedAssertOneThread::fatal_different() VL_MT_SAFE {
 #endif
 
 //===========================================================================
+// VlDeleter:: Methods
+
+void VlDeleter::deleteAll() {
+    VerilatedLockGuard lock{m_mutex};
+    while (!m_newGarbage.empty()) {
+        std::swap(m_newGarbage, m_toDelete);
+        lock.unlock();
+        for (VlClass* const objp : m_toDelete) delete objp;
+        m_toDelete.clear();
+        lock.lock();
+    }
```

Comment:
```suggestion
    while (!m_newGarbage.empty()) {
        {
            VerilatedLockGuard lock{m_mutex};
            std::swap(m_newGarbage, m_toDelete);
        }
        for (VlClass* const objp : m_toDelete) delete objp;
        m_toDelete.clear();
    }
```

Reply:
Is this change under the (currently correct) assumption that this function is called from one thread only? If so, we'll need an assertion for that. Otherwise, I think this introduces the risk of two or more threads entering the loop body, with the first thread to get the lock iterating `m_toDelete`, and the second thread swapping the vectors again while the first one is iterating.

Reply:
And more critically checking empty() isn't guaranteed to be a threadsafe operation.

Reply:
Ok, ignore me please.